resources/stateresourcemanager.go

```diff
@@ -27,24 +27,57 @@ var resourceRegistry = map[string]reflect.Type{
 	AutoSubnetResource: reflect.TypeOf(AutoSubnetCfgResource{}),
 }
 
-// EtcdResourceManager implements the core.ResourceManager interface.
-// It manages the resource in a logically centralized manner using serialized
-// writes to a etcd based datastore.
-type EtcdResourceManager struct {
-	//XXX: should be '*drivers.EtcdStateDriver', but leaving is
-	//core.StateDriver to get tests going and until the netmaster
-	//is changed to pickup the resource-manager from config
-	Etcd core.StateDriver
+// StateResourceManager implements the core.ResourceManager interface.
+// It manages the resources in a logically centralized manner using serialized
+// writes to underlying state store.
+type StateResourceManager struct {
+	stateDriver core.StateDriver
+}
+
+var gStateResourceManager *StateResourceManager
+
+// NewStateResourceManager instantiates a state based resource manager
+func NewStateResourceManager(sd core.StateDriver) (*StateResourceManager, error) {
+	if gStateResourceManager != nil {
+		return nil, core.Errorf("state-based resource manager instance already exists.")
+	}
+
+	gStateResourceManager = &StateResourceManager{stateDriver: sd}
```

Comment:
mutex is not needed right now as the assumption is that this is a singleton instance that is created just once by the main routine at startup and is never released/modified by other go routines. We can revisit if this assumption changes.